src/models.py

```diff
@@ -744,7 +743,7 @@ def forward(self, task, batch, predict=False):
                 self.utilization(get_batch_utilization(batch["input"]))
         if isinstance(task, SingleClassificationTask):
             out = self._single_sentence_forward(batch, task, predict)
-        elif isinstance(task, GLUEDiagnosticTask):
+        elif task.is_diagnostic:
```

Comment:
As above, I'd vote to revert this change. For deciding whether to train, it makes sense to check a simple Boolean that we can easily reuse for new tasks. But for deciding which model code to run, we actually need to know something about tho data structures used for the task, so I think it's right to check isinstance or something similarly specific.